pyqt-apps/siriushla/as_di_bpms/main.py

```diff
@@ -101,3 +152,84 @@ def setupui(self):
             self.layoutv.addWidget(grpbx)
             self.layoutv.addSpacing(20)
             self.layoutv.addStretch()
+
+    def _amcFRU(self, i):
+        pbt = QPushButton('AFC Settings')
+        Window = create_window_from_widget(
+            AFCAdvancedSettings, title=self.bpm+': AFC Settings')
+        util.connect_window(
+            pbt, Window, parent=self, prefix=self.prefix, bpm=self.bpm)
+
+        pv_list_fru = afcs_pvs.AFCv3_1_PV_LIST['FRU']
+        grbx = CustomGroupBox('AFC Details', self)
+        grbx_lay = QHBoxLayout(grbx)
+        grbx_lay.setSpacing(20)
+
+        grbx_lay.addWidget(pbt)
+
+        if self.bpm[:2] == 'TS' or self.bpm[:2] == 'TB':
+            cmd_button = SiriusPushButton(
+                self, label='Reset', icon=qta.icon('mdi.restart'),
+                init_channel=f"IA-20RaBPMTL:{afcs_pvs.DIS}-AMC-{i}:{pv_list_fru['SoftRst']}"
+                )
+        elif self.bpm[:2] == 'BO':
+            bo_num = int(self.bpm.split('-')[1][:2])
+            if bo_num in afcs_pvs.BO_N:
+                area_index = afcs_pvs.BO_N.index(bo_num) + 1
+                area_str = f"{area_index:02}"
+                cmd_button = SiriusPushButton(
+                    self, label='Reset', icon=qta.icon('mdi.restart'),
+                    init_channel=f"IA-{area_str}RaBPM:{afcs_pvs.DIS}-AMC-{i}:{pv_list_fru['SoftRst']}"
+                    )
```

Comment:
there is several blocks of code repeated here, you can use the `if` block only for defining the variables `subsection` to be used in the `init_channel` parameter in `IA-{subsection}RaBPM` and then have a block of code that creates your button.